### ISSUE TITLE
Wrong results after deploying keras model to Coral Dev Board

### ISSUE THREAD

**Issue by Kampi** [google-coral/edgetpu#402]
I try to run some examples out of my Deep Learning Books with my Coral Dev Board. Please take a look at the attached files for the source codes.

1. I run `python3 shallownet_train.py --dataset datasets/animals --model model/Animals.h5 --labels model/Animals.pickle` to train the model and got the following results:

```
[INFO] Serializing network...
[INFO] Evaluating network...
              precision    recall  f1-score   support

         cat       0.49      0.54      0.51       249
         dog       0.43      0.35      0.39       262
       panda       0.63      0.67      0.65       239

    accuracy                           0.52       750
   macro avg       0.51      0.52      0.52       750
weighted avg       0.51      0.52      0.51       750
```

2. I run `python3 shallownet_load.py --dataset datasets/animals --model model/Animals.h5 --labels model/Animals.pickle` to test the model:

```
datasets/animals/cats/cats_00919.jpg
cats
datasets/animals/dogs/dogs_00585.jpg
dogs
datasets/animals/cats/cats_00120.jpg
cats
datasets/animals/cats/cats_00285.jpg
cats
datasets/animals/panda/panda_00214.jpg
cats
datasets/animals/dogs/dogs_00774.jpg
dogs
datasets/animals/cats/cats_00352.jpg
cats
datasets/animals/cats/cats_00786.jpg
cats
datasets/animals/panda/panda_00465.jpg
panda
datasets/animals/cats/cats_00476.jpg
cats
```

3. I run `python3 Quantize.py --dataset datasets/animals --model model/Animals.h5 --tflite-model model/Animals.tflite` to generate a TensorFlow light model

```
[INFO] Loading images...
[INFO] processed 100/100
fully_quantize: 0, inference_type: 6, input_inference_type: 0, output_inference_type: 0
[INFO] Serializing the TFLite model...
```

4. I use my Docker container to compile the model for the Edge TPU `docker run -it --rm -v /var/development/Edge\ Example/model:/home/edgetpu edgetpu_compiler`

```
root@b34eb9af3deb:/home/edgetpu# edgetpu_compiler Animals.tflite
Edge TPU Compiler version 15.0.340273435

Model compiled successfully in 218 ms.

Input model: Animals.tflite
Input size: 100.15KiB
Output model: Animals_edgetpu.tflite
Output size: 564.53KiB
On-chip memory used for caching model parameters: 3.75KiB
On-chip memory remaining for caching model parameters: 7.85MiB
Off-chip memory used for streaming uncached model parameters: 512.06KiB
Number of Edge TPU subgraphs: 1
Total number of operations: 6
Operation log: Animals_edgetpu.log
See the operation log file for individual operation details.
```

5. I copy `Animals.pickle`, `Animals_edgetpu.tflite` and the images from (2) on my Coral Dev Board
6. I run `python3 Classifier.py --dataset datasets/animals --model model/Animals_edgetpu.tflite --labels model/Animals.pickle` with three different images and the result is always the same (and wrong):

```
mendel@edge:~/ShallowNet$ python3 Classifier.py --dataset datasets/animals --model model/Animals_edgetpu.tflite --labels model/Animals.pickle --image datasets/animals/cats/cats_00919.jpg
[INFO] Loading class labels...
[INFO] Loading Coral model...
[INFO] Making predictions...
----INFERENCE TIME----
Note: The first inference on Edge TPU is slow because it includes loading the model into Edge TPU memory.
4.7ms
-------RESULTS--------
Class(id=0, score=0.99609375)
cats
Class(id=1, score=0.0)
dogs
Class(id=2, score=0.0)
panda
mendel@edge:~/ShallowNet$ python3 Classifier.py --dataset datasets/animals --model model/Animals_edgetpu.tflite --labels model/Animals.pickle --image datasets/animals/dogs/dogs_00585.jpg
[INFO] Loading class labels...
[INFO] Loading Coral model...
[INFO] Making predictions...
----INFERENCE TIME----
Note: The first inference on Edge TPU is slow because it includes loading the model into Edge TPU memory.
3.1ms
-------RESULTS--------
Class(id=0, score=0.99609375)
cats
Class(id=1, score=0.0)
dogs
Class(id=2, score=0.0)
panda
mendel@edge:~/ShallowNet$ python3 Classifier.py --dataset datasets/animals --model model/Animals_edgetpu.tflite --labels model/Animals.pickle --image datasets/animals/panda/panda_00465.jpg
[INFO] Loading class labels...
[INFO] Loading Coral model...
[INFO] Making predictions...
----INFERENCE TIME----
Note: The first inference on Edge TPU is slow because it includes loading the model into Edge TPU memory.
3.4ms
-------RESULTS--------
Class(id=0, score=0.99609375)
cats
Class(id=1, score=0.0)
dogs
Class(id=2, score=0.0)
panda
```

[Code.zip](https://github.com/google-coral/edgetpu/files/6676430/Code.zip)
[Models.zip](https://github.com/google-coral/edgetpu/files/6718329/Models.zip)

**Comment by Naveen-Dodda** [comment]
@Kampi 

Can you attach your both edgetpu.tflite and .tflite models along with examples images that you tested. I Could not find them in the Code.zip

**Comment by Kampi** [comment]
> @Kampi
> 
> Can you attach your both edgetpu.tflite and .tflite models along with examples images that you tested. I Could not find them in the Code.zip

Please take a look at my updated question.

**Comment by hjonnala** [comment]
@Kampi i haven't found the example images in the code.zip and models.zip. Please send us the example images that you tested.

**Comment by Kampi** [comment]
> @Kampi i haven't found the example images in the code.zip and models.zip. Please send us the example images that you tested.

Hi,

I can not add the images as ZIP container, so you have to download them from [my NextCloud](https://cloud.server-kampert.de/s/r56MpTnzManw9Bo).

Please note that I have used specific images on my Edge device (see my initial post for the image names).

**Comment by Kampi** [comment]
@hjonnala 

I can not add the images as ZIP container, so you have to download them from [my NextCloud](https://cloud.server-kampert.de/s/r56MpTnzManw9Bo).

Please note that I have used specific images on my Edge device (see my initial post for the image names).

**Comment by hjonnala** [comment]
Hi,

I am unable to run the shallownet_train.py due to pyimagesearch dependency.

So I have used one of our tutorial to build the model using mobile net with the provided datasets.

Tutorials: https://github.com/google-coral/tutorials

Please check these files in the attachment:
1. mobilenet_v2_1.0_224_quant_edgetpu.tflite (converted edgetpu model)
2. python code to model and generate edgetpu file (402_edgetpu_mobile_net.ipynb)
3. results.txt
![Screenshot from 2021-07-11 07-52-26](https://user-images.githubusercontent.com/86132058/125199853-fe662200-e21c-11eb-98c0-a3bcb826f2ac.png)


[Models.tar.gz](https://github.com/google-coral/edgetpu/files/6797109/Models.tar.gz)

**Comment by Kampi** [comment]
Hi,

thank you for the link. I will try it out in the next few days. I think the tutorial is also a good base for using custom models instead of `MobileNetV2`.
